File: backend/src/main/java/com/dcm/backend/entities/FileHeader.java
package com.dcm.backend.entities;

import com.dcm.backend.annotations.LogIgnore;
import com.dcm.backend.enumeration.Status;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Collection;

@Entity
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class FileHeader {

    @Id
    @LogIgnore
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String folder;

    private String filename;

    @LogIgnore
    private String thumbnailName;

    private String description;

    private String version;

    private Status status;

    private LocalDate date;

    private String type;

    private Long size;

    @ManyToMany
    private Collection<Keyword> keywords;
    
}

File: frontend/src/app/models/FileHeader.ts
export interface FileHeader {
  id: string;
  folder: string;
  filename: string;
  thumbnailName: string;
  thumbnail: string | undefined;
  description: string;
  version: string;
  status: string;
  date: string;
  type: string;
  size: number;
  printableSize: string;
  keywords: string[];
}

export function convertSizeToPrintable(size: number): string {
  if (size < 1024) {
    return size + ' bytes';
  }
  if (size < 1024 * 1024) {
    return (size / 1024).toFixed(2) + ' KB';
  }
  return (size / 1024 / 1024).toFixed(2) + ' MB';
}
